src/test/java/teammates/ui/webapi/GetNotificationsActionTest.java

```diff
@@ -167,11 +181,58 @@ public void testExecute_invalidUserType_shouldFail() {
                 String.valueOf(true));
     }
 
-    private void verifyNotificationEquals(NotificationAttributes expected, NotificationData actual) {
+    @Test
+    public void testExecute_withFalseIsFetchingAll_shouldReturnUnreadNotifications() {
+        ______TS("Request to fetch unread notification only");
+
+        InstructorAttributes instructor = typicalBundle.instructors.get("instructor1OfCourse1");
+        loginAsInstructor(instructor.getGoogleId());
+        Set<String> readNotificationsId = typicalBundle.accounts.get("instructor1OfCourse1").getReadNotifications().keySet();
+
+        String[] requestParams = new String[] {
+                Const.ParamsNames.NOTIFICATION_TARGET_USER, NotificationTargetUser.INSTRUCTOR.toString(),
+                Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL, String.valueOf(false),
+        };
+
+        GetNotificationsAction action = getAction(requestParams);
+        JsonResult jsonResult = getJsonResult(action);
+
+        NotificationsData output = (NotificationsData) jsonResult.getOutput();
+        List<NotificationData> notifications = output.getNotifications();
+        verifyDoesNotContainNotifications(notifications, readNotificationsId);
+    }
+
+    @Test
+    public void testExecute_withoutIsFetchingAll_shouldReturnUnreadNotifications() {
+        ______TS("Request without isfetchingall");
```

Comment:
Just to clarify, is `isFetchingAll` null only for admin get request?

Reply:
Currently this parameter is optional. Its absence means a default false value.